packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SurroundingPairScopeHandler/RangeLookupList.ts

```diff
@@ -0,0 +1,49 @@
+import type { Range } from "@cursorless/common";
+
+/**
+ * An iterator that allows for efficient lookup of ranges that contain a search item.
+ * The items must be sorted in document order.
+ */
+export class RangeLookupList<T extends { range: Range }> {
```

Comment:
Is this correct? If you accept my name change, don't forget to rename doc

```suggestion
/**
 * Given a list of ranges (the haystack), allows the client to search for a sequence of ranges (the needles).
 * Has the following requirements:
 * - the haystack must be sorted in document order
 * - **the needles must be in document order as well**. This enables us to avoid backtracking as you search for a sequence of items.
 * - the haystack entries must not overlap. Adjacent is fine
 */
export class OneWayRangeFinder<T extends { range: Range }> {
```